### PR TITLE
AJ-1009 create snapshot ref on import

### PR DESCRIPTION
**Pull request by dvoet**
Reminder:

PRs merged into main will not automatically generate a PR in https://github.com/broadinstitute/terra-helmfile to update the WDS image deployed to kubernetes - this action will need to be triggered manually by running the following github action: https://github.com/DataBiosphere/terra-workspace-data-service/actions/workflows/tag.yml. 
After your manually trigger the github action (and it completes with no errors), you must go to [this](https://github.com/broadinstitute/terra-helmfile) repo and verify that this generated a PR that merged succesfully.
The terra-helmfile PR merge will then generate a PR in https://github.com/DataBiosphere/leonardo.  This may or may not automerge; be sure to watch it to ensure it merges.

### CODE REVIEW

File: service/src/main/java/org/databiosphere/workspacedataservice/controller/DataRepoController.java [hunk 1]
```diff
@@ -13,21 +14,26 @@
...
+                                                             @PathVariable("snapshotId") UUID snapshotId) {
+        // getSnapshot will throw exception is caller does not have access
+        SnapshotModel snapshot = dataRepoDao.getSnapshot(snapshotId);
+
+        // createDataRepoSnapshotReference is required to setup policy and will throw exception if policy conflicts
+        workspaceManagerDao.createDataRepoSnapshotReference(snapshot);
+
+        // TODO do the import
+
+        return new ResponseEntity<>(HttpStatus.ACCEPTED);
```
Comment (dvoet) [inline service/src/main/java/org/databiosphere/workspacedataservice/controller/DataRepoController.java:37]: I wanted the name of the snapshot so I changed `hasSnapshotPermission` to `getSnapshot` and fiddled with the exception handling

File: service/src/main/java/org/databiosphere/workspacedataservice/controller/DataRepoController.java [hunk 2]
```diff
@@ -13,21 +14,26 @@
...
-            return new ResponseEntity<>(HttpStatus.FORBIDDEN);
-        }
+                                                             @PathVariable("snapshotId") UUID snapshotId) {
+        // getSnapshot will throw exception is caller does not have access
+        SnapshotModel snapshot = dataRepoDao.getSnapshot(snapshotId);
+
+        // createDataRepoSnapshotReference is required to setup policy and will throw exception if policy conflicts
+        workspaceManagerDao.createDataRepoSnapshotReference(snapshot);
+
+        // TODO do the import
```
Comment (davidangb) [inline service/src/main/java/org/databiosphere/workspacedataservice/controller/DataRepoController.java:35]: probably want to move this logic to a service class once we actually start doing the import - don't need to change anything in this PR, just thinking out loud

File: service/src/main/java/org/databiosphere/workspacedataservice/workspacemanager/WorkspaceManagerDao.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    try {
+      String timeStamp = new SimpleDateFormat("yyyyMMddHHmmss").format(new java.util.Date());
+      resourceApi.createDataRepoSnapshotReference(new CreateDataRepoSnapshotReferenceRequestBody()
+          .snapshot(
+              new DataRepoSnapshotAttributes()
+                  .instanceName(INSTANCE_NAME)
+                  .snapshot(snapshotModel.getId().toString()))
+              .metadata(
+                  new ReferenceResourceCommonFields()
+                      .cloningInstructions(CloningInstructionsEnum.LINK_REFERENCE)
```
Comment (davidangb) [inline service/src/main/java/org/databiosphere/workspacedataservice/workspacemanager/WorkspaceManagerDao.java:39]: thank you!

**Review by davidangb — APPROVED** [review]
lgtm, thank you!

File: service/src/main/java/org/databiosphere/workspacedataservice/controller/DataRepoController.java (hunk 2 above)
Comment (dvoet) [reply service/src/main/java/org/databiosphere/workspacedataservice/controller/DataRepoController.java:35]: definitely agree

**Review by calypsomatic — APPROVED** [review]
Looks great, thank you!

**Pull request merged**